test/test-server/cloud_kafka_handlers.go

```diff
@@ -47,8 +46,8 @@ var KafkaQuotas = []*kafkaquotasv1.KafkaQuotasV1ClientQuota{
 func handleKafkaClientQuota(t *testing.T) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		quotaId := mux.Vars(r)["id"]
-		if i := slices.IndexFunc(KafkaQuotas, func(quota *kafkaquotasv1.KafkaQuotasV1ClientQuota) bool { return quota.GetId() == quotaId }); i != -1 {
-			quota := KafkaQuotas[i]
+		if i := getV2Index(kafkaQuotas, quotaId); i != -1 {
```

Comment:
What does `getV2Index()` mean?

Reply:
I just couldn't think of a better name. It's to get the index of the entry in the array that matches the id entered. I wrote it specifically for things in ccloud-sdk-go-v2 which all have a `GetId()` method, hence why I slapped v2 in the name.

Reply:
Is your goal with this to add state to the mock backend? How about we make a generic `type mockDatabase []T` that has functions like `Add()`, `Lookup()`, and `Delete()`?

Reply:
Kind of. I initially added state before I realized that this was incredibly annoying: having state in our test backend means that running a specific test with `INTEGRATION_TEST_ARGS` can fail if updates, posts, and deletes that normally run beforehand don't get run. It also obfuscates what's actually changing when the output has `updated-name` but the test only updated a different field...

So what I'm trying to do here is adding a consistent initial state, so that describe/list/delete are all consistent. And none of the delete/patch/post actually change anything, so you always have the same set of resources to work with when you add additional integration tests, and you don't have to update unrelated golden files when you re-arrange or remove integration tests.

Side note: I didn't bother trying to do this for api-key because we have a state for that one with the keystore...